packages/stakers/src/set/ensureStakerPkgsNetworkConfig.ts

```diff
@@ -0,0 +1,202 @@
+ * - **selected** execution client
+ * - **selected** consensus client
+ */
+export async function ensureStakerPkgsNetworkConfig(
+  network: Network
+): Promise<void> {
+  const stakerConfig = getStakerConfigByNetwork(network);
+  const { executionClients, consensusClients } =
+    getStakerDnpNamesByNetwork(network);
+
+  for (const { dnpName, clientsToCheck, alias } of [
+    {
+      dnpName: stakerConfig.executionClient,
+      clientsToCheck: executionClients,
+      alias: getStakerFullnodeAlias(network, stakerConfig.executionClient),
+    },
+    {
+      dnpName: stakerConfig.consensusClient,
+      clientsToCheck: consensusClients,
+      alias: null,
+    },
+  ]) {
+    // skip if no client selected
+    if (!dnpName) continue;
+
+    const pkg = await listPackageNoThrow({
+      dnpName: dnpName,
+    });
+    // skip if pkg not installed
+    if (!pkg) continue;
+
+    // Disconnect and remove staker network from other clients
+    await disconnectAndRemoveStakerNetworkFromCompose(clientsToCheck);
+
+    // Add staker network and fullnode alias if available to client compose file
+    await connectPkgToStakerNetwork(
+      params.DOCKER_STAKER_NETWORK_NAME,
+      pkg.containers,
+      alias
+    );
+    // Connect client to staker network
+    addStakerNetworkToCompose(dnpName, alias);
+  }
+}
+
+async function disconnectAndRemoveStakerNetworkFromCompose(
+  clientsToCheck: readonly string[]
+): Promise<void> {
+  for (const client of clientsToCheck) {
+    const clientPkg = await listPackageNoThrow({
+      dnpName: client,
+    });
+    // skip if client not installed
+    if (!clientPkg) continue;
+
+    // Disconnect client from staker network
+    await disconnectConnectedPkgFromStakerNetwork(
+      params.DOCKER_STAKER_NETWORK_NAME,
+      clientPkg.containers
+    );
+    removeStakerNetworkFromCompose(client);
+  }
+}
+
+function removeStakerNetworkFromCompose(dnpName: string): void {
+  // remove from compose network
+  const compose = new ComposeFileEditor(dnpName, false);
+  delete compose.compose.networks?.[params.DOCKER_STAKER_NETWORK_NAME];
+  compose.write();
+  // remove from compose service network
+  for (const [serviceName, service] of Object.entries(
+    compose.compose.services
+  )) {
+    const composeService = new ComposeFileEditor(dnpName, false);
+    // network declared in array format is not supported
+    if (Array.isArray(service.networks)) {
+      logs.warn(
+        `Service ${serviceName} in ${dnpName} has a network declared in array format, skipping`
+      );
+      continue;
+    }
+    composeService
+      .services()
+      // eslint-disable-next-line no-unexpected-multiline
+      [serviceName].removeNetwork(params.DOCKER_STAKER_NETWORK_NAME);
+    composeService.write();
+  }
+}
+
+async function disconnectConnectedPkgFromStakerNetwork(
+  networkName: string,
+  pkgContainers: PackageContainer[]
+): Promise<void> {
+  const connectedContainers = pkgContainers
```

Comment:
Not related to this exact line, but we need to be sure that a container connected to the stakers network that becomes stopped will not re-connect once it is started

Reply:
thats ensure with the compose file